modules/Bio/EnsEMBL/Compara/DnaFrag.pm

```diff
@@ -196,12 +196,22 @@ sub new {
 sub new_from_Slice {
     my ($class, $slice, $genome_db) = @_;
 
-    my ($attrib) = @{ $slice->get_all_Attributes('codon_table') };
     my $codon_table_id;
-    $codon_table_id = $attrib->value() if $attrib;
-    my ($seq_loc) = @{ $slice->get_all_Attributes('sequence_location') };
     my $sequence_location;
-    $sequence_location = $seq_loc->value() if $seq_loc;
+    my $is_reference;
+
+    if ($slice->{'attributes'}) {
+        $codon_table_id    = $slice->{'attributes'}->{'codon_table'};
+        $sequence_location = $slice->{'attributes'}->{'sequence_location'};
+        $is_reference      = exists $slice->{'attributes'}->{'non_ref'} ? 0 : 1;
+
+    } else {
+        my ($codon_table_attrib) = @{ $slice->get_all_Attributes('codon_table') };
```

Comment:
Again, out of curiosity, what is the difference between this line and
```
my $codon_table_attrib = $slice->get_all_Attributes('codon_table');
```

Reply:
`get_all_Attributes` returns an arrayref of Attribute objects, so not your solution but maybe this
```
        my $codon_table_attrib = $slice->get_all_Attributes('codon_table')->[0];
```